packages/material-ui/src/useMediaQuery/useMediaQuery.js

```diff
@@ -1,5 +1,9 @@
 import * as React from 'react';
 import { getThemeProps, useTheme } from '@material-ui/styles';
+import useEnhancedEffect from '../utils/useEnhancedEffect';
+
+const useEnhancedEffectNoTests =
```

Comment:
To me, this looks like we are pretending that our tests are working. Isn't it better to just remove the tests that are failing instead of using different logic when we are in a test environment?

Reply:
I believe that the test case is passing and what's failing is the security we have in place to fail when a warning is raised (by React, in our case).

Reply:
Got ya, then maybe I would suggest to better expect this warning in the tests. If the warning is the only thing, I am ok with even leaving as it is, but at least let's add a comment above the code explaining why we have this condition.

Reply:
Oh why not, we do that in NoSsr, @marthaerm do you want to work on this?

Reply:
Yes! Of course! I'll start working on it! Thanks for the review